src/db.rs

```diff
@@ -2906,12 +2942,220 @@ pub struct LiveFile {
+        static EMPTY: [u8; 0] = [];
+        let empty_ptr = EMPTY.as_ptr() as *const libc::c_char;
+
+        unsafe {
+            let live_files = LiveFilesGuard(ffi::rocksdb_livefiles_create());
+
+            for file in files {
+                let live_file = LiveFileGuard(ffi::rocksdb_livefile_create());
+                ffi::rocksdb_livefile_set_level(live_file.0, file.level);
+
+                // SAFETY: C strings are copied inside the FFI layer so do not need to be kept alive
+                let c_cf_name = CString::new(file.column_family_name.clone()).map_err(|err| {
+                    Error::new(format!("Unable to convert column family to CString: {err}"))
+                })?;
+                ffi::rocksdb_livefile_set_column_family_name(live_file.0, c_cf_name.as_ptr());
+
+                let c_name = CString::new(file.name.clone()).map_err(|err| {
+                    Error::new(format!("Unable to convert file name to CString: {err}"))
+                })?;
+                ffi::rocksdb_livefile_set_name(live_file.0, c_name.as_ptr());
+
+                let c_directory = CString::new(file.directory.clone()).map_err(|err| {
+                    Error::new(format!("Unable to convert directory to CString: {err}"))
+                })?;
+                ffi::rocksdb_livefile_set_directory(live_file.0, c_directory.as_ptr());
+
+                ffi::rocksdb_livefile_set_size(live_file.0, file.size);
+
+                let (start_key_ptr, start_key_len) = match &file.start_key {
+                    None => (empty_ptr, 0),
+                    Some(key) => (key.as_ptr() as *const libc::c_char, key.len()),
+                };
+                ffi::rocksdb_livefile_set_smallest_key(live_file.0, start_key_ptr, start_key_len);
+
+                let (largest_key_ptr, largest_key_len) = match &file.end_key {
+                    None => (empty_ptr, 0),
+                    Some(key) => (key.as_ptr() as *const libc::c_char, key.len()),
+                };
+                ffi::rocksdb_livefile_set_largest_key(
+                    live_file.0,
+                    largest_key_ptr,
+                    largest_key_len,
+                );
+                ffi::rocksdb_livefile_set_smallest_seqno(live_file.0, file.smallest_seqno);
+                ffi::rocksdb_livefile_set_largest_seqno(live_file.0, file.largest_seqno);
+                ffi::rocksdb_livefile_set_num_entries(live_file.0, file.num_entries);
+                ffi::rocksdb_livefile_set_num_deletions(live_file.0, file.num_deletions);
+
+                // moves ownership of live_files into live_file
+                ffi::rocksdb_livefiles_add(live_files.0, live_file.into_raw());
+            }
+
+            // moves ownership of live_files into inner
+            ffi::rocksdb_export_import_files_metadata_set_files(self.inner, live_files.into_raw());
+            Ok(())
+        }
+    }
+}
+
+impl Default for ExportImportFilesMetaData {
+    fn default() -> Self {
```

Comment:
would be good to do:
```
        let inner = unsafe { ffi::rocksdb_export_import_files_metadata_create() };
        assert!(
            !inner.is_null(),
            "Could not create rocksdb_export_import_files_metadata_t"
        );
        Self { inner }
```
to ensure the pointer is non-null, matching the style of other wrappers